File: C:C++/FizzBuzz.c
// Marisa Pugliese
// Solution to FizzBuzz in C

// Instructions to compile (using gcc compiler) and run in the command line:
// Be sure to cd into this directory
// gcc -o fizzbuzz fizzbuzz.c
// ./fizzbuzz

#include <stdio.h>
#include <stdlib.h>

void fizzbuzz(int max, int fizz, int buzz){
    for (int i = 0; i <= max; i++){
        char* f = "Fizz";
        char* b = "Buzz";
        int flag = 0;
        if (i%(fizz*buzz) == 0){
            printf("%s%s", f, b);
            printf("\n");
            flag = 1;
        }
        else if (i%fizz == 0){
            printf("%s", f);
            printf("\n");
            flag = 1;
        }
        else if (i%buzz == 0){
            printf("%s", b);
            printf("\n");
            flag = 1;
        }
        if (flag == 0){
            printf("%d", i);
            printf("\n");
        }
    }
}

int main(){
    // Feel free to change max, the upper bound on the sequence.
    int max = 50;
    fizzbuzz(max, 3, 5);
    return 0;
}

File: Python/FizzBuzz.py
# FizzBuzz.py
# Marisa Pugliese
# Here is my solution to FizzBuzz writen in Python

# Instructions to compile and run in the command line
# Be sure cd into this directory
# python FizzBuzz.py

def fizz_buzz(max, fizz, buzz):
    for i in range(max):
        if i%(fizz*buzz) == 0:
            print("FizzBuzz")
        elif i%fizz == 0:
            print("Fizz")
        elif i%buzz == 0:
            print("Buzz")
        else:
            print(i)

def main():
    max = 30
    fizz_buzz(max, 3, 5)

main()
